pkg/registry/core/service/storage/patch_rest.go

```diff
@@ -0,0 +1,28 @@
+package storage
+
+import (
+	"context"
+
+	"k8s.io/apiserver/pkg/warning"
+	api "k8s.io/kubernetes/pkg/apis/core"
+)
+
+// OpenShift 4.8 and 4.9 only - BZ 2045576
+// In kube 1.21 and 1.22 (OCP 4.8 and 4.9), the apiserver would default the value of `ipFamilyPolicy` to `RequireDualStack`
+// if you created a Service with two `ipFamilies` or two `clusterIPs` but no explicitly-specified `ipFamilyPolicy`.
+// In 1.23 / 4.10, you MUST explicitly specify either "ipFamilyPolicy: PreferDualStack" or "ipFamilyPolicy: RequireDualStack"
+// for the service to be valid.
+// Emit a warning in 4.8 and 4.9 if such services are created or updated.
+// Unfortunately, this cannot be done with a mutating or validating admission controller webhook because
+// tryDefaultValidateServiceClusterIPFields will set service.Spec.IPFamilyPolicy before the webhooks have a chance to inspect
+// it. Therefore, we implement this downstream only warning message in OpenShift's kube-apiserver.
```

Comment:
@sttts the field population happens on the Rest Create and Update hooks that AFAIK run after the admission , so this statement is not true, am I wrong?